lib/jquery.ui/jquery.ui.unitsuggester.js

```diff
@@ -0,0 +1,307 @@
+( function( $, util, mw ) {
+	'use strict';
+
+var PARENT = $.ui.suggester;
+
+/**
+ * @class jQuery.ui.languagesuggester
+ * @extends jQuery.ui.suggester
+ * @licence GNU GPL v2+
+ * @author Thiemo Mättig
+ * @author Jonas Kress
+ *
+ * @constructor
+ */
+
+$.widget( 'wikibase.unitsuggester', PARENT, {
+
+	/**
+	 * Options
+	 * @property {Object}
+	 */
+	options: {
+		url: 'https://www.wikidata.org/w/api.php',
+		language: mw && mw.config && mw.config.get( 'wgUserLanguage' ),
+		timeout: 8000
+	},
+
+	/**
+	 * @property {number}
+	 * @private
+	 */
+	_searchTimeoutHandle: null,
+
+	/**
+	 * @property {string}
+	 * @private
+	 */
+	_selectedUrl: null,
+
+	/**
+	 * Caches retrieved results.
+	 * @property {Object} [_cache={}]
+	 * @private
+	 */
+	_cache: {},
+
+	/**
+	 * @inheritdoc
+	 * @protected
+	 */
+	_create: function() {
+		var self = this;
+
+		this._cache = {};
+		this.options.source = this._initDefaultSource();
+
+		PARENT.prototype._create.call( this );
+
+		this.element
+			.addClass( 'ui-unitsuggester-input' )
+			.prop( 'dir', $( document ).prop( 'dir' ) );
+
+		this.options.menu.element.addClass( 'ui-unitsuggester-list' );
+
+		this.element
+		.off( 'blur' )
+		.on( 'eachchange.' + this.widgetName, function( event ) {
+			self._search( event );
+			self._trigger( 'change' );
+		} );
+	},
+
+	/**
+	 * @inheritdoc
+	 */
+	destroy: function() {
+		this.element.removeClass( 'ui-unitsuggester-input' );
+
+		this._cache = {};
+
+		PARENT.prototype.destroy.call( this );
+	},
+
+	/**
+	 * @protected
+	 *
+	 * @param {jQuery.Event} event
+	 */
+	_search: function( event ) {
+		var self = this;
+		this._cache = {};
+		this._selectedUrl = null;
+
+		self._term = self.element.val();
+
+		clearTimeout( this._searchTimeoutHandle );
+		this._searchTimeoutHandle = setTimeout( function() {
+			self.search()
+			.done( function( suggestions ) {
```

Comment:
Why did you removed the `if( !suggestions.length || self.element.val() !== requestTerm )` check?